### ISSUE TITLE
Intermittent failures on dbt jobs due to Fabric connection error

### ISSUE THREAD

**Issue by sophiad96** [microsoft/dbt-fabric#112]
A dbt user is experiencing intermittent job failures due to the following error occurring on different models:

`('IMC06', '[IMC06] [Microsoft][ODBC Driver 18 for SQL Server]The connection is broken and recovery is not possible. The connection is marked by the client driver as unrecoverable. No attempt was made to restore the connection. (0) (SQLExecDirectW)')`

This is causing frequent failures in job runs for this user, which impacts their production data.

**Comment by prdpsvs** [comment]
@sophiad96 , This is a transient error. The adapter is not using ConnectRetryCount property, which is exposed by ODBC. ODBC driver will attempt to automatically reestablish the connection if ConnectRetryCount and ConnectRetryInterval are provided. Let me add these options to the connection string to recover the connection. I will do a patch release 1.7.4 this week to handle this issue. I will use retries property in profiles.yaml file for ConnectRetryCount and add ConnectRetryInterval to profiles.yaml default to 2 seconds.
 
However, adapter is currently retrying connections manually. Are you providing retry option in the profiles.yaml? If not, please provide that and see if it helps.

**Comment by sophiad96** [comment]
@prdpsvs thank you for this information! The user currently has their retries set to 5.

**Comment by sophiad96** [comment]
@prdpsvs Curious if we have any updates to share with the customer?

**Comment by prdpsvs** [comment]
@sophiad96 , I came to know that there are connection drops reported by a few customers who are not using dbt as well. This issue is related to Fabric DW itself. However, the latest patch 1.7.4 will be available by tomorrow with ConnectRetryCount in the connection string.

**Comment by SaintofSpadesSOA** [comment]
When will this patch be available in dbt Cloud?

**Comment by amychen1776** [comment]
@SaintofSpadesSOA this patch should already be in dbt Cloud - has it resolved your issues?

**Comment by SaintofSpadesSOA** [comment]
@amychen1776  No. We have been experiencing the same errors since the aforementioned release above.  Retries are still set at 5. It's does not always occur, but is still seen quite often.

**Comment by SaintofSpadesSOA** [comment]
@prdpsvs  Is there any timeline to the issue being fixed by Microsoft pertaining to the Fabric DW? Have any of the affected customers you've previously mentioned found any workarounds to the issue? I encounter this issue daily which has pretty much thrown all reliability/dependability out the window with this tool.

**Comment by prdpsvs** [comment]
@TheBishop-98 , Are you still facing this issue? the adapter has a few configurations included in V1.8.9 including connection pooling and retrycount and retryconnect interval.

**Comment by prdpsvs** [comment]
TDS endpoint V3 is released and adapter is now handling connection and query failures with a retry. I am closing this due to inactivity.

Please re-open if you still see the issue.
